nvme.c

```diff
@@ -7281,6 +7186,7 @@ static int submit_io(int opcode, char *command, const char *desc, int argc, char
 	}
 
 	if (strlen(cfg.data)) {
+		close(dfd);
```

Comment:
In the last year there were many changes in this code path. It seems it is very hard to get the various conditions right so that there is no leak etc. My approach might look a bit ugly but given the fact that we had some many bugs and ugly conditions to figure out if a file descriptor needs to be close or not I'd prefer my stupid way but easy to follow approach.

Reply:
You were right :) I'd changed the way you suggested and it's better.